src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
+
+  decreaseHealth(num) {
+    const posOfAll = Animal.alive.indexOf(this);
+
+    this.health -= num;
+
+    if (this.health <= 0 && posOfAll > -1) {
+      Animal.alive.splice(posOfAll, 1);
+    }
+  }
```

Comment:
Move this logic into Carnivore bite method.
Also use Array.filter instead of splice

Reply:
@IvanFesenko Can you describe why splice method is not appropriate for this?

Reply:
If you have any questions - ask them in fe_chat